backend/src/api/UserController.ts

```diff
@@ -338,4 +341,33 @@ export default class UserController {
             return response.error('error', `Could not update bio`, 500);
         }
     }
+
+    async barmaliniPassword(request: APIRequest<BarmaliniPasswordRequest>, response: APIResponse<BarmaliniPasswordResponse>) {
```

Comment:
Add a check that current user is already a barmalini.

Reply:
It's covered by user restrictions, but won't hurt to add an explicit check.